Library/Homebrew/dev-cmd/pr-publish.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+require "utils/github"
+
+module Homebrew
+  module_function
+
+  def pr_publish_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `pr-publish` <pull_request>
+
+        Publishes bottles for a pull request with GitHub Actions.
+        Requires write access to the repository.
+      EOS
+      switch :verbose
+    end
+  end
+
+  def pr_publish
+    pr_publish_args.parse
+
+    odie "You need to specify a pull request number!" if Homebrew.args.named.empty?
+
+    args.named.each do |arg|
+      arg = "#{CoreTap.instance.default_remote}/pull/#{arg}" if arg.to_i.positive?
+      url_match = arg.match HOMEBREW_PULL_OR_COMMIT_URL_REGEX
+      _, user, repo, issue = *url_match
+      tap = Tap.fetch(user, repo) if repo.match?(HOMEBREW_OFFICIAL_REPO_PREFIXES_REGEX)
+      odie "Not a GitHub pull request: #{arg}" unless issue
+      ohai "Dispatching #{tap} pull request ##{issue}"
+      GitHub.dispatch(user, repo, "Publish ##{issue}", pull_request: issue)
```

Comment:
Could this (or a line after) output a URL that you can use to check progress? If not possible: no worries.

Reply:
This unfortunately returns 204 No Content. We'd have to find it using another API call.

Reply:
😭. Even if it outputs the full PR URL somewhere that might be nice. Where does the "action" show up; just in the actions tab on the repo?

Reply:
Yup, just in the Actions tab. You can filter by actor and workflow type though which can help. In Linuxbrew/core we have @BrewTestBot ping the requestor when the upload job fails. This could be added to the workflow in Homebrew/core:

https://github.com/Homebrew/linuxbrew-core/blob/3d460c215425e7d1547865ad0c2f5b4b7bce7dd7/.github/workflows/upload-bottles.yml#L165-L173
https://github.com/Homebrew/linuxbrew-core/pull/19902#issuecomment-601881442

Reply:
Yeh, this would be great to have there too 👍